Plugins/BuildServerIntegration/TeamCityIntegration/TeamCityAdapter.cs

```diff
@@ -412,18 +412,28 @@ private Task<Stream> GetStreamFromHttpResponseAsync(Task<HttpResponseMessage> ta
 
                 var buildServerCredentials = _buildServerWatcher.GetBuildServerCredentials(this, true);
                 var useBuildServerCredentials = buildServerCredentials is not null
-                                                && !buildServerCredentials.UseGuestAccess
-                                                && (string.IsNullOrWhiteSpace(buildServerCredentials.Username) && string.IsNullOrWhiteSpace(buildServerCredentials.Password));
+                                                && buildServerCredentials.BuildServerCredentialsType == BuildServerCredentialsType.UsernameAndPassword
+                                                && !string.IsNullOrWhiteSpace(buildServerCredentials.Username) && !string.IsNullOrWhiteSpace(buildServerCredentials.Password);
                 if (useBuildServerCredentials)
                 {
                     UpdateHttpClientOptionsCredentialsAuth(buildServerCredentials!);
-                    return GetStreamAsync(restServicePath, cancellationToken);
                 }
-                else
+
+                var useBuildServerBearerToken = buildServerCredentials is not null
```

Comment:
We shouldn't use `var` for the intrinsic types and when the type can't be easily inferred from the context.